java/core/src/java/org/apache/orc/impl/StringHashTableDictionary.java

```diff
@@ -75,18 +75,19 @@ public StringHashTableDictionary(int initialCapacity, float loadFactor) {
     this.capacity = initialCapacity;
     this.loadFactor = loadFactor;
     this.keyOffsets = new DynamicIntArray(initialCapacity);
-    initHashBuckets(initialCapacity);
+    initHashBuckets(false);
     this.threshold = (int)Math.min(initialCapacity * loadFactor, MAX_ARRAY_SIZE + 1);
   }
 
-  private void initHashBuckets(int capacity) {
-    DynamicIntArray[] buckets = new DynamicIntArray[capacity];
-    for (int i = 0; i < capacity; i++) {
+  private void initHashBuckets(final boolean reinit) {
+    final DynamicIntArray[] newBuckets =
+        (reinit) ? this.hashBuckets : new DynamicIntArray[this.capacity];
```

Comment:
Do we need to call some kind of `clear` or null out the current elements to ensure they're able to be GC'd as soon as possible? I've noticed that this new `hash` dictionary uses more memory than the `rbtree` dictionary in some tests I'm working on.

Reply:
As soon as each element in `buckets` is overwritten with a new bucket, it becomes available for GC.  There is no value in writing a null value and the immediately replacing it with a new bucket.